src/lib.rs

```diff
@@ -176,25 +230,55 @@ impl ProcessManager {
 				let backoff = backoff.as_millis() as u64;
 				let jittered_delay: u64 = rand::thread_rng().gen_range(0..backoff);
 				let backoff = Duration::from_millis(
-					2_u64.saturating_pow(process_data.restarts as u32) * jittered_delay,
+					2_u64
+						.saturating_pow(process_data.restarts as u32)
+						.saturating_mul(jittered_delay),
 				);
+				info!(
+					"sleeping for {}ms before restarting process {} (restart {})",
+					backoff.as_millis(),
+					process_data.process.executable,
+					process_data.restarts
+				);
+
 				tokio::time::sleep(backoff).await;
 			}
 			RestartMode::Delayed(backoff) => {
+				info!(
+					"sleeping for {}ms before restarting process {} (restart {})",
+					backoff.as_millis(),
+					process_data.process.executable,
+					process_data.restarts
+				);
 				tokio::time::sleep(backoff).await;
 			}
 			RestartMode::Instant => {}
 		}
 
-		let command = Command::new(&process_data.process.executable)
-			.args(&process_data.process.args)
-			.envs(process_data.process.env.clone())
-			.stdout(Stdio::piped())
-			.stderr(Stdio::piped())
-			.stdin(Stdio::null())
-			.kill_on_drop(true)
-			.spawn()
-			.map_err(Error::Process)?;
+		let command = unsafe {
+			Command::new(&process_data.process.executable)
+				.args(&process_data.process.args)
+				.envs(process_data.process.env.clone())
+				.stdout(Stdio::piped())
+				.stderr(Stdio::piped())
+				.stdin(Stdio::piped())
+				.kill_on_drop(true)
+				.pre_exec({
```

Comment:
Seems a bit odd to use a block here. I was confused before I realized the first part wasn't part of the closure. Might be clearer to move `let fd_list = ...` to before `let command`?

But more importantly, this implementation looks like it should be correct and sound. Just calling `mark_as_not_cloexec()` should hopefully be sound, while doing much more in `pre_exec` may not be.  :+1: